crates/polars-ops/src/chunked_array/scatter.rs

```diff
@@ -174,11 +161,14 @@ impl ChunkedSet<bool> for &BooleanChunked {
         V: IntoIterator<Item = Option<bool>>,
     {
         check_bounds(idx, self.len() as IdxSize)?;
-        check_sorted(idx)?;
         let mut ca_iter = self.into_iter().enumerate();
         let mut builder = BooleanChunkedBuilder::new(self.name().clone(), self.len());
 
-        for (current_idx, current_value) in idx.iter().zip(values) {
+        let mut idx_values_iter: Vec<(_, _)> = idx.iter().zip(values).collect();
+        // Sort by index's elements
+        idx_values_iter.sort_by_key(|&(idx_item, _)| idx_item);
```

Comment:
I don't think we should always sort. Only if the data isn't sorted.

So we can `check_sorted` and only sort on the error branch.

Reply:
From the docs...

https://doc.rust-lang.org/stable/std/primitive.slice.html#current-implementation-10

> ..., achieving linear time on fully sorted and reversed inputs.


----------

Happy to implement the check on `idx` before calling `sort_by_key()`.

The docs also state that there is a possibility for some memory allocation depending on the input size. So if we iterate over the index values, it should avoid the possible memory allocation.

Reply:
I've implemented a helper method that is leveraged in both locations.

Resolving